src/p5xr/core/raycasting.js

```diff
@@ -0,0 +1,189 @@
+
+  let leftPMatrixInverse = new p5.Matrix();
+  leftPMatrixInverse.invert(p5xr.instance.viewer.leftPMatrix.copy());
+  leftPMatrixInverse.transpose(leftPMatrixInverse);
+  leftPMatrixInverse = leftPMatrixInverse.mat4;
+
+  let leftDirectionCopy = leftDirection.copy();
+  leftDirection.x = leftPMatrixInverse[0] * leftDirectionCopy.x + leftPMatrixInverse[1] * leftDirectionCopy.y + leftPMatrixInverse[2] * leftDirectionCopy.z + leftPMatrixInverse[3];
+  leftDirection.y = leftPMatrixInverse[4] * leftDirectionCopy.x + leftPMatrixInverse[5] * leftDirectionCopy.y + leftPMatrixInverse[6] * leftDirectionCopy.z + leftPMatrixInverse[7];
+  leftDirection.normalize();
+
+  // get ray direction from right eye
+  let rightDirection = new p5.Vector(screenX, screenY, -1);
+  
+  let rightPMatrixInverse = new p5.Matrix();
+  rightPMatrixInverse.invert(p5xr.instance.viewer.rightPMatrix.copy());
+  rightPMatrixInverse.transpose(rightPMatrixInverse);
+  rightPMatrixInverse = rightPMatrixInverse.mat4;
+
+  let rightDirectionCopy = rightDirection.copy();
+  rightDirection.x = rightPMatrixInverse[0] * rightDirectionCopy.x + rightPMatrixInverse[1] * rightDirectionCopy.y + rightPMatrixInverse[2] * rightDirectionCopy.z + rightPMatrixInverse[3];
+  rightDirection.y = rightPMatrixInverse[4] * rightDirectionCopy.x + rightPMatrixInverse[5] * rightDirectionCopy.y + rightPMatrixInverse[6] * rightDirectionCopy.z + rightPMatrixInverse[7];
+  rightDirection.normalize();
+
+  // combine both ray directions
+  ray.direction = p5.Vector.add(leftDirection, rightDirection).normalize();
+
+  return ray;
+}
+
+p5.prototype.intersectsSphere = function() {
+  let radius = arguments[0];
+  let ray = {
+    origin: null,
+    direction: null
+  };
+  if(arguments.length !== 2 || !arguments[1].hasOwnProperty('origin')) {
+    let screenX = arguments[1] || 0, screenY = arguments[2] || 0;
+    ray = getRayFromScreen(screenX, screenY);
+  }
+  else {
+    ray.origin = arguments[1].origin.copy();
+    ray.direction = arguments[1].direction.copy();
+  }
+
+  if(ray === null)
+    return false;
+    
+  // sphere in View space
+  let uMVMatrix = p5.instance._renderer.uMVMatrix.copy();
+  uMVMatrix.transpose(uMVMatrix);
+  uMVMatrix = uMVMatrix.mat4;
+    
+  let sphereCenter = new p5.Vector(0, 0, 0);
+  sphereCenter.x = uMVMatrix[3];
+  sphereCenter.y = uMVMatrix[7];
+  sphereCenter.z = uMVMatrix[11];
+
+  if(p5.Vector.sub(ray.origin, sphereCenter).mag() <= radius) {
+    return true;
+  }
+
+  // check if sphere is in front of ray
+  if(p5.Vector.dot(p5.Vector.sub(sphereCenter, ray.origin), ray.direction) < 0) {
+    return false;
+  }
+
+  let sphereToRayOrigin = p5.Vector.sub(ray.origin, sphereCenter);
+  let b = 2 * p5.Vector.dot(ray.direction, sphereToRayOrigin);
+  let c = p5.Vector.mag(sphereToRayOrigin) * p5.Vector.mag(sphereToRayOrigin) - radius * radius;
+
+  let det = b * b - 4 * c;
+
+  return det >= 0;
+};
+
+p5.prototype.intersectsBox = function() {
```

Comment:
Moving [a discussion ](https://github.com/stalgiag/p5.xr/pull/48/files#r305584870)to this PR:

Main issue is how to setup the parameters for `intersectsBox`. All of the following should be possible:
```
intersectsBox(10) //checks for intersection between a ray originating at screen 0, 0 and a box with dimensions 10, 10, 10
intersectsBox(10, 20, 10, 0, 0) //checks for intersection btwn a ray originating at screen 0, 0 and a box with dimensions 10, 20, 10 -- 
```
Note that the screenX and screenY must be specified in the last example. This is unfortunate but straightforward for the time being.

Additionally these should work
```
intersectsBox(10, ray) 
intersectsBox(10, 20, 10, ray)

Reply:
I have added the default values.